chatgpt-shell.el

```diff
@@ -272,6 +278,26 @@ Used by `chatgpt-shell--send-input's call."
                           chatgpt-shell--prompt-internal)))
     (nreverse result)))
 
+(defun chatgpt-shell--write-output-to-log-buffer (output)
+  "Write curl process OUTPUT to log buffer.
+
+Create the log buffer if it does not exist.  Pretty print output
+if `json' is available."
+  (let ((buffer (get-buffer chatgpt-shell--log-buffer-name)))
+    (unless buffer
+      ;; Create buffer
+      (setq buffer (get-buffer-create chatgpt-shell--log-buffer-name))
+      (with-current-buffer buffer
+        ;; Use `js-json-mode' if available, fall back to `js-mode'.
+        (funcall (if (fboundp #'js-json-mode)
```

Comment:
nit: Slight preference for:

    (if (fboundp #'js-json-mode)
        (js-json-mode +1)
      (js-mode +1))